JM/CF/Scripts/3_Game/CommunityFramework/Spline/CF_SplinePoint.c

```diff
@@ -0,0 +1,246 @@
+
+		SetSpeedCoef(1.0);
+		SetRotationDerived(true);
+	}
+	
+	void ~CF_SplinePoint()
+	{
+	}
+
+	void SetSpline(CF_Spline spline)
+	{
+		m_Spline = spline;
+	}
+
+	CF_Spline GetSpline()
+	{
+		return m_Spline;
+	}
+	
+	void SetPosition(vector position, bool imd = true)
+	{
+		m_Position = position;
+
+		if (imd && m_Spline) m_Spline._Update();
+	}
+	
+	vector GetPosition()
+	{
+		return m_Position;
+	}
+
+	void CalculateTangent(bool imd = true)
+	{
+		float diff = GetEnd();
+		if (m_Prev) diff = GetEnd() - m_Prev.GetStart();
+		
+		vector pv = vector.Zero;
+		vector nv = vector.Zero;
+
+		if (m_Next) nv = vector.Direction(m_Next.GetPosition(), m_Position);
+		if (m_Prev) pv = vector.Direction(m_Position, m_Prev.GetPosition());
+		
+		if (!m_Next && m_Prev) nv = vector.Direction(m_Prev.GetPosition(), m_Position);
+		if (!m_Prev && m_Next) pv = vector.Direction(m_Position, m_Next.GetPosition());
+
+		m_Tangent = (pv + nv) * -(1.0 - m_Spline.GetTension()) * (1.0 / Math.Max(0.0001, diff));
+
+		if (imd && m_Spline) m_Spline._Update();
+	}
+
+	void SetTangent(vector tangent, bool imd = true)
+	{
+		m_Tangent = tangent;
+		m_Type = CF_SplinePointType.CUBIC_USER;
+
+		if (imd && m_Spline) m_Spline._Update();
+	}
+	
+	vector GetTangent()
+	{
+		return m_Tangent;
+	}
+	
+	void SetDirection(vector direction, vector up = "0 1 0")
+	{
+		vector mat[4];
+		Math3D.DirectionAndUpMatrix(direction, up, mat);
+		m_Orientation = Math3D.MatrixToAngles(mat);
+		Math3D.MatrixToQuat(mat, m_Rotation);
+
+		SetRotationDerived(false);
+	}
+	
+	void SetOrientation(vector orientation)
+	{
+		m_Orientation = orientation;
+		
+		vector mat[3];
+		m_Orientation.RotationMatrixFromAngles(mat);
+		Math3D.MatrixToQuat(mat, m_Rotation);
+
+		SetRotationDerived(false);
+	}
+	
+	vector GetOrientation()
+	{
+		return m_Orientation;
+	}
+
+	void SetType(CF_SplinePointType type, bool imd = true)
+	{
+		m_Type = type;
+
+		if (imd && m_Spline) m_Spline._Update();
+	}
+
+	CF_SplinePointType GetType()
+	{
+		return m_Type;
+	}
+
+	void SetRotationDerived(bool derived)
+	{
+		m_DerivedRotation = derived;
+	}
+
+	bool IsRotationDerived()
+	{
+		return m_DerivedRotation;
+	}
+
+	void SetSpeedCoef(float speedCoef)
+	{
+		m_SpeedCoef = speedCoef;
+		if (m_SpeedCoef <= 0.0) m_SpeedCoef = 1.0;
+	}
+
+	float GetSpeedCoef()
+	{
+		return m_SpeedCoef;
+	}
+
+	float GetLength()
+	{
+		return m_Length;
+	}
+
+	float GetStart()
+	{
+		return m_Start;
+	}
+
+	float GetEnd()
+	{
+		return m_Start + m_Length;
+	}
+
+	void Get(float delta, out vector position, out vector orientation)
+	{
+		position = m_Position;
+		orientation = m_Orientation;
+		
+		switch (m_Next.m_Type)
+		{
+			case CF_SplinePointType.LINEAR:
+				LinInterp(delta, position, orientation);
+				break;
+			case CF_SplinePointType.CUBIC:
+			case CF_SplinePointType.CUBIC_USER:
+				CubicInterp(delta, position, orientation);
+				break;
+		}
+	}
+
+	void LinInterp(float delta, out vector position, out vector orientation)
```

Comment:
Rename `LinInterp` to the commonly used term `Lerp`

Reply:
Name the other one `Cerp`?

Named it like that to keep the name similar to `CubicInterp`

Reply:
I commented on the other name: https://github.com/Jacob-Mango/DayZ-Community-Framework/pull/71#discussion_r689532668

Linear interpolation is either written out or `Lerp`, no matter which language/framework/engine.